src/domain/accounts/index.types.d.ts

```diff
@@ -46,6 +46,21 @@ type AccountStatusHistory = Array<{
   comment?: string
 }>
 
+type AccountCustomFieldValues =
+  | string
+  | number
+  | boolean
+  | string[]
+  | number[]
+  | boolean[]
+
+type AccountCustomFields = {
+  readonly accountId: AccountId
+  modifiedByUserId: UserId
+  customFields: { [k: string]: AccountCustomFieldValues }
+  createdAt: Date
```

Comment:
shouldn't those be read only also?

for `createdAt` and `modifiedByUserId`, those values should be immutable also. 

for `customFields` it seems we are creating a new object when we're changing those property so it could be readonly also?

Reply:
Updated here https://github.com/GaloyMoney/galoy/pull/1465/commits/2d687e1d6ee24f6539fe767272ac77a138f8084a but customFields cant be readonly because need to be updated (default values use case)